examples/ConversionForm/CollectionInput.tsx

```diff
@@ -0,0 +1,124 @@
+import { map, reduce } from "lodash";
+import React from "react";
+import BaseInput from "./BaseInput";
+
+interface CollectionInputProps {
+    handler: (path: string[], key: string, value: any) => void;
+    id: string;
+    templateData: { [key: string]: any };
+    name: string;
+    path: string[];
+    parameter: { [key: string]: any };
+    dataType: string;
+}
+
+class CollectionInput extends React.Component<CollectionInputProps> {
+    constructor(props: CollectionInputProps) {
+        super(props);
+        this.state = {};
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(key, index, type, targetValue) {
```

Comment:
this function is kind of begging for unit tests as documentation of how it's supposed to work in practice